lib/src/diagnostics/inspector_service.dart

```diff
@@ -0,0 +1,29 @@
+import 'package:rohd/rohd.dart';
+
+/// `ModuleTree` implements the Singleton design pattern
+/// to ensure there is only one instance of it during runtime.
+///
+/// This class is used to maintain a tree-like structure
+/// for managing modules in an application.
+class ModuleTree {
+  /// Private constructor used to initialize the Singleton instance.
+  ModuleTree._();
+
+  /// Singleton instance of `ModuleTree`.
+  ///
+  /// Always returns the same instance of `ModuleTree`.
+  static ModuleTree get instance => _instance;
```

Comment:
I remember there are some issues on the DevTool extension side. It can't seems to obtain the object from static variable object. 

```dart
final treeInstance = await rohdControllerEval.evalInstance(
        'ModuleTree.instance.hierarchyJSON',
        isAlive: evalDisposable
);
```

So I refer to the example in the `foo` example, it seems that they use singleton to get things work. 

https://github.com/flutter/devtools/blob/master/packages/devtools_extensions/example/foo/packages/foo/lib/src/foo_controller.dart

```dart
static FooController get instance => _instance;
static final _instance = FooController._();
```